### ISSUE TITLE
Improve GC performance when loading a snapshot

### ISSUE THREAD

**Issue by lemmih** [ChainSafe/forest#2686]
**Issue summary**

The semi-space GC loads snapshots into the "new" space rather than the "old" space. This immediately triggers a garbage collection run which isn't necessary and doesn't actually copy anything (yet still wastes a lot of resources). We should load snapshots into the "old" space.

**Other information and links**

**Comment by lemmih** [comment]
Depends on #2638

**Comment by lemmih** [comment]
I don't think that is sustainable. Conceptually, we really do want the snapshot data to be placed in the "old" space.

**Comment by LesnyRumcajs** [comment]
The new, shiny GC (https://github.com/ChainSafe/forest/pull/3504) is on it's way, which makes this issue no longer relevant.
